core/journal/journal.go

```diff
@@ -27,6 +28,7 @@ type JournalEntry struct {
 type Journal struct {
 	entries    []JournalEntry
 	EntryLimit int
+	mutex      sync.Mutex
```

Comment:
`sync.RWMutex` is preferred as we need to synchronize the reading as well. With `sync.Mutex` readers need to wait for each other.

Reply:
Actually forget about the read lock, as too many read locking could block writes and in turns block proxying with current implementation. journal write need to made async via a channel or some other way first to avoid this issue